src/Ast.ml

```diff
@@ -1292,8 +1292,7 @@ end = struct
               Ppat_construct
                 ({txt= Lident "::"}, Some {ppat_desc= Ppat_tuple [_; tl]})
           }
-      , Ppat_construct ({txt= Lident "::"}, _) )
-      when tl == pat ->
+      , Ppat_construct ({txt= Lident "::"}, _) ) when tl == pat ->
```

Comment:
There shouldn't be (significant) formatting changes like this when the option is not enabled. It is definitely intended that when a pattern does not fit on a line, the `when` clause should start at the beginning of a line.